tsk/fs/ext2fs.c

```diff
@@ -55,24 +55,32 @@ debug_print_buf(unsigned char *buf, int len)
 
 /** \internal
     test_root - tests to see if a is power of b
-    Adapted from E2fsprogs sparse.c
-    Super blocks are only in block groups that are powers of 3,5, and 7
     @param a the number being investigated
     @param b the root
     @return 1 if a is a power of b, otherwise 0
 */
 static uint8_t
 test_root(uint32_t a, uint32_t b)
 {
-    if (a == 0)
-        return 1;
-    while (1) {
-        if (a == 1)
-            return 1;
-        if (a % b)
-            return 0;
-        a = a / b;
+    if (a == 0) {
+        return (b == 0);
+    }
+    else if (b == 0) {
+        return 0;
     }
+    else if (a == 1) {
+        return (b == 1);
```

Comment:
Do we care about the fact that any value for b raised to the zero power would be 1?

Reply:
Updated